test/test-framework/pkg/controller/memcached/memcached_controller.go

```diff
@@ -180,10 +180,6 @@ func (r *ReconcileMemcached) deploymentForMemcached(m *cachev1alpha1.Memcached)
 	replicas := m.Spec.Size
 
 	dep := &appsv1.Deployment{
-		TypeMeta: metav1.TypeMeta{
```

Comment:
Was this manually removed?

Reply:
We should NOT add the TypeMeta. This data is filled when the object is initialized.

Reply:
Yeah I think this is safe to remove ([see our current controller scaffold](https://github.com/operator-framework/operator-sdk/blob/9bc2d204564cc6b8883f511b7f4f3f6efa901764/internal/scaffold/controller_kind.go#L265-L281)). IIRC there was a bug in controller-runtime at one point that required us to include this that has since been fixed.

Reply:
HI @estroz and @joelanford,
 
We should remove it from all places. Note that is not a good practice at all add it.
We removed from samples in the past. However, since I note that has other places as well, then I remove it from here and I will do a specific pr for that in order to make clear the change and motivation.